src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesSignedCommits.java

```diff
@@ -0,0 +1,162 @@
+package com.sap.sgs.phosphor.fosstars.data.github;
+
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.USES_VERIFIED_SIGNED_COMMITS;
+
+import com.sap.sgs.phosphor.fosstars.data.UserCallback;
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.value.BooleanValue;
+import com.sap.sgs.phosphor.fosstars.tool.YesNoSkipQuestion;
+import com.sap.sgs.phosphor.fosstars.tool.YesNoSkipQuestion.Answer;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+import org.json.JSONObject;
+import org.kohsuke.github.GHCommit;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * This data provider tries to figure out if a project uses signed commits. First, it checks if a
+ * project belongs to an organization and checks if any commits are signed or verified. It gathers
+ * the list of commits within one year from the current date. Then it calculates density ratio
+ * between number of signed commits vs unsigned commits. If the density ratio >= 1, then we set the
+ * flag = true else flag = false. Foundation. Next, it tries to ask a user if {@link UserCallback}
+ * is available.
+ */
+public class UsesSignedCommits extends AbstractGitHubDataProvider {
+
+  /**
+   * 1 year in millis.
+   */
+  private static final long DELTA = 365 * 24 * 60 * 60 * 1000L;
+
+  /**
+   * The percentage threshold of number of signed commits against number of total commits.
+   */
+  private static final double SIGNED_COMMIT_PRECENTAGE_THRESHOLD = 0.4;
```

Comment:
I would suggest to use much higher threshold value, let's say 90% or even more. What do you think?

Reply:
If I understand correctly, `0.4` doesn't mean percentage, it means a fraction. You didn't mean 0.4%, did you? Please drop the "PRECENTAGE" from the constant name or update the constant value (by the way, there is a typo in PRECENTAGE).

Reply:
@artem-smotrakov , I ran a quick test on `https://github.com/spring-projects/spring-integration` as the input for the past year:

- Number of signed commits: 24
- Total number of commits: 493

Density percent : 4.8681 %. 

Looking at this I am thinking to keep the SIGNED_COMMIT_PERCENTAGE_THRESHOLD = 10. What do you think?

Reply:
@sourabhsparkala I don't think we should align the threshold value with one (or even many) projects.

The new feature answers the question "Does the project uses verified signed commits?". If I am given a project, and if I am asked this question, then I would ask myself "Are most of commits in the projects signed?". If I see that that only 4.86% of commits are signed, I would answer no to the question. Would you answer yes?

I think that "most of commits" means at least 90%. I would even say 95%. If a project require commits to be signed, then an unsigned commit should be a rare exception.

Reply:
Yes I agree. with your comment.